firefox-ios/Client/Frontend/TabContentsScripts/UserScriptManager.swift

```diff
@@ -39,54 +39,87 @@ class UserScriptManager: FeatureFlaggable {
-                    source: wrappedSource,
-                    injectionTime: injectionTime,
-                    forMainFrameOnly: mainFrameOnly
-                )
+            if let userScript = UserScriptManager.getUserScriptForFrame(name: name,
+                                                                        injectionTime: injectionTime,
+                                                                        mainFrameOnly: mainFrameOnly) {
                 compiledUserScripts[name] = userScript
             }
 
             // Autofill scripts
             let autofillName = "Autofill\(name)"
-            if let autofillScriptCompatPath = Bundle.main.path(
-                forResource: autofillName, ofType: "js"),
-                let source = try? NSString(
-                    contentsOfFile: autofillScriptCompatPath,
-                    encoding: String.Encoding.utf8.rawValue) as String {
-                let wrappedSource = "(function() { const APP_ID_TOKEN = '\(UserScriptManager.appIdToken)'; \(source) })()"
-                let userScript = WKUserScript.createInDefaultContentWorld(
-                    source: wrappedSource,
-                    injectionTime: injectionTime,
-                    forMainFrameOnly: mainFrameOnly)
+            if let userScript = UserScriptManager.getUserScriptForAutofill(autofillName: autofillName,
+                                                                           injectionTime: injectionTime,
+                                                                           mainFrameOnly: mainFrameOnly) {
                 compiledUserScripts[autofillName] = userScript
             }
 
             let webcompatName = "Webcompat\(name)"
-            if let webCompatPath = Bundle.main.path(
-                forResource: webcompatName,
-                ofType: "js"
-            ),
-               let source = try? NSString(
-                contentsOfFile: webCompatPath,
-                encoding: String.Encoding.utf8.rawValue
-               ) as String {
-                let wrappedSource = "(function() { const APP_ID_TOKEN = '\(UserScriptManager.appIdToken)'; \(source) })()"
-                let userScript = WKUserScript.createInPageContentWorld(
-                    source: wrappedSource,
-                    injectionTime: injectionTime,
-                    forMainFrameOnly: mainFrameOnly
-                )
+            if let userScript = UserScriptManager.getUserScriptForWebcompat(webcompatName: webcompatName,
+                                                                            injectionTime: injectionTime,
+                                                                            mainFrameOnly: mainFrameOnly) {
                 compiledUserScripts[webcompatName] = userScript
             }
         }
 
         self.compiledUserScripts = compiledUserScripts
     }
 
+    private static func getUserScriptForFrame(name: String,
```

Comment:
I have needed to create the following functions as `static` to use them in the `init` function.